tests/torch/accuracy_aware_training/test_training_loop.py

```diff
@@ -119,3 +128,134 @@ def configure_optimizers_fn():
+            with torch.no_grad():
+                for _ in range(num_steps):
+                    x, y_gt = next(train_loader)
+                    y = model(x)
+                    loss += F.mse_loss(y.sum(), y_gt)
+        return 1 - loss.item()
+
+    config = get_quantization_config_without_range_init(LeNet.INPUT_SIZE[-1])
+
+    params = {
+        "maximal_accuracy_degradation": max_accuracy_degradation,
+        "maximal_total_epochs": maximal_total_epochs,
+    }
+    accuracy_aware_config = {
+        "accuracy_aware_training": {
+            "mode": "early_exit",
+            "params": params
+        }
+    }
+
+    config.update(accuracy_aware_config)
+
+    model, train_loader, compression_ctrl = create_finetuned_lenet_model_and_dataloader(config,
+                                                                                        validate_fn,
+                                                                                        init_finetuning_steps)
+
+    def train_fn(compression_ctrl, model, epoch, optimizer, lr_scheduler,
+                 train_loader=train_loader):
+        with set_torch_seed():
+            train_loader = iter(train_loader)
+            for _ in range(num_steps):
+                compression_ctrl.scheduler.step()
+                optimizer.zero_grad()
+                x, y_gt = next(train_loader)
+                y = model(x)
+                loss = F.mse_loss(y.sum(), y_gt)
+                loss.backward()
+                optimizer.step()
+
+    def configure_optimizers_fn():
+        optimizer = SGD(model.parameters(), lr=learning_rate)
+        return optimizer, None
+
+    early_stopping_training_loop = EarlyExitCompressionTrainingLoop(config, compression_ctrl)
+    model = early_stopping_training_loop.run(model,
+                                             train_epoch_fn=train_fn,
+                                             validate_fn=partial(validate_fn, train_loader=train_loader),
+                                             configure_optimizers_fn=configure_optimizers_fn)
+    original_model_accuracy = model.original_model_accuracy
+    compressed_model_accuracy = validate_fn(model, train_loader=train_loader)
+
+    assert (original_model_accuracy - compressed_model_accuracy) * 100 <= max_accuracy_degradation
+
+
+# mock validation func
+
+# 6 epoch give correct val metric and
+
+# assert (stop_epoch_num == 6)
+
+@pytest.mark.parametrize(
+    ('max_accuracy_degradation', 'exit_epoch_number'),
+    ((1.0, 6), (30.0, 10))
+)
+def test_early_exit_with_mock_validation(max_accuracy_degradation, exit_epoch_number,
+                                         maximal_total_epochs=100):
+    def mock_validate_fn(model, epoch_counter=None, init_step=False, epoch=0):
+        if epoch_counter is None:
+            epoch_counter = epoch
+        original_metric = 0.85
+        if init_step:
+            return original_metric
+        else:
+            epoch_counter[0] = epoch
```

Comment:
In this case it is possible to use closure instead of having to pass an additional mutable arg to the function. Just define `epoch_counter = 0` above your `mock_validate_fn` and reference `epoch_counter` in your function code.

Reply:
It was my first idea, I don't know why but it doesn't work.